src/typeahead/typeahead.directive.ts

```diff
@@ -486,13 +486,17 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
 
             return typeahead
               .pipe(
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore - mystery error: Error: src/typeahead/typeahead.directive.ts:489:17 - error TS2554: Expected 0 arguments, but got 2.
```

Comment:
@adventmedia You can get rid of those `@ts-ignore` by casting `typeahead` as `Observable<TypeaheadOption[]>`: 
```ts
const typeahead = (isObservable(this.typeahead) ? this.typeahead : from(this.typeahead)) as Observable<TypeaheadOption[]>;

return typeahead
  .pipe(
    filter((option: TypeaheadOption) => {
      return !!option && this.testMatch(this.normalizeOption(option), normalizedQuery);
    }),
    toArray()
  );
```

Reply:
requested change made